File: src/main/java/app/application/utils/ComponentUtils.java
package app.application.utils;

import javafx.fxml.FXMLLoader;
import javafx.scene.Parent;
import lombok.SneakyThrows;

public class ComponentUtils {

	private ComponentUtils(){
		throw new IllegalStateException("Utility Class");
	}

	@SneakyThrows
	public static <T extends Parent> void loadComponent(T component){
		FXMLLoader loader = new FXMLLoader();
		loader.setRoot(component);
		loader.setControllerFactory(aClass -> component);
		String fileName = "/components/" + component.getClass().getSimpleName() + ".fxml";
		loader.load(component.getClass().getResourceAsStream(fileName));
	}

}

File: src/main/kotlin/app/application/utils/ComponentUtils.kt
package app.application.utils

import javafx.fxml.FXMLLoader
import javafx.scene.Parent
import javafx.util.Callback
import lombok.SneakyThrows

class ComponentUtils private constructor() {
    init {
        throw IllegalStateException("Utility Class")
    }

    companion object {
        @JvmStatic
				@SneakyThrows
        fun <T : Parent?> loadComponent(component: T?) {
            val loader = FXMLLoader()
            loader.setRoot(component)
            loader.controllerFactory = Callback { component }
            val fileName = "/components/" + component!!::class.simpleName + ".fxml"
            loader.load<Any>(component.javaClass.getResourceAsStream(fileName))
        }
    }
}